.jenkins/utils.groovy

```diff
@@ -0,0 +1,66 @@
+// A simple utility to mark the build as pending on Github.
+def startBuild(String context)
+{
+  step([
+      $class: "GitHubCommitStatusSetter",
+      reposSource: [$class: "ManuallyEnteredRepositorySource",
+                    url: "https://github.com/mlpack/mlpack"],
+      contextSource: [$class: "ManuallyEnteredCommitContextSource",
+                      context: context ],
+      errorHandlers: [[$class: "ChangingBuildStatusErrorHandler",
+                       result: "UNSTABLE"]],
+      statusResultSource: [$class: "ConditionalStatusResultSource",
+                           results: [[$class: "AnyBuildResult",
+                                      message: "Building...",
+                                      state: "PENDING"]]]
+  ]);
+}
+
+// A simple utility to set the build status on Github for a commit.
+def setBuildStatus(Map paramsMap)
+{
+  // Extract arguments from the map.
+  def result = paramsMap.result;
+  def context = paramsMap.context;
+  def successMessage = paramsMap.successMessage;
```

Comment:
Groovy looks interesting

Reply:
I guess it's not bad, but I am tired of having so many different languages... I would have rather just had a shell script.